modules/renter/files.go

```diff
@@ -201,25 +201,34 @@ func (r *Renter) DeleteFile(nickname string) error {
 
 // FileList returns all of the files that the renter has.
 func (r *Renter) FileList() []modules.FileInfo {
+	var files []*file
 	lockID := r.mu.RLock()
-	defer r.mu.RUnlock(lockID)
-
-	files := make([]modules.FileInfo, 0, len(r.files))
 	for _, f := range r.files {
+		files = append(files, f)
+	}
+	r.mu.RUnlock(lockID)
+
+	isOffline := func(id types.FileContractID) bool {
+		return r.hostContractor.IsOffline(id)
+
+	}
+
+	var fileList []modules.FileInfo
+	for _, f := range files {
 		f.mu.RLock()
 		renewing := true
-		files = append(files, modules.FileInfo{
+		fileList = append(fileList, modules.FileInfo{
 			SiaPath:        f.name,
 			Filesize:       f.size,
-			Available:      f.available(r.hostContractor.IsOffline),
-			Redundancy:     f.redundancy(r.hostContractor.IsOffline),
 			Renewing:       renewing,
+			Available:      f.available(isOffline),
+			Redundancy:     f.redundancy(isOffline),
 			UploadProgress: f.uploadProgress(),
 			Expiration:     f.expiration(),
 		})
 		f.mu.RUnlock()
```

Comment:
This means that the hostContractor and the file can't be locked at the same time from other functions. Are sure that's the right approach?

I'm still not super happy with the way this is constructed.